From 70de8ffb592d7d801736c9af1012bce453b493d6 Mon Sep 17 00:00:00 2001
From: MengLi <meng.li@windriver.com>
Date: Wed, 4 Jul 2018 15:42:39 +0800
Subject: [PATCH 1336/1345] arm64: dts: marvell: improve sata dts noed for
 armada platform.

The sdk kernel patches that are related with sata dts node are
applied for admada-cp110.dtsi and admada-cp110-1.dtsi file.
But in mainline kernel, they should be applied for
admada-cp110-master.dtsi and admada-cp110-slave.dtsi file.
So, merge code related with sata dts node into the 2 dtsi files.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040.dtsi       |    4 ++--
 .../boot/dts/marvell/armada-cp110-master.dtsi      |   17 +++++++++++++++--
 .../arm64/boot/dts/marvell/armada-cp110-slave.dtsi |   17 +++++++++++++++--
 3 files changed, 32 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040.dtsi b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
index 4c9afb7..2bab406 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
@@ -66,7 +66,7 @@
 			smmu: iommu@5000000 {
 				mmu-masters = <&cpm_usb3_0 0x483>,
 					      <&cpm_usb3_1 0x484>,
-					      <&sata 0x485>,
+					      <&cpm_sata0 0x485>,
 					      <&cp_xor0 0x489>,
 					      <&cp_xor1 0x48a>;
 			};
@@ -114,7 +114,7 @@
 			ppv22@000000 {
 				dma-coherent;
 			};
-			sata: sata@540000 {
+			cpm_sata0: sata@540000 {
 				#stream-id-cells = <1>;
 			};
 			cp_xor0: cpxor0@6a0000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
index 8120ba8..b5ab005 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
@@ -141,12 +141,25 @@
 			};
 
 			cpm_sata0: sata@540000 {
-				compatible = "marvell,armada-8k-ahci",
-					     "generic-ahci";
+				compatible = "marvell,armada-cp110-ahci";
 				reg = <0x540000 0x30000>;
+				dma-coherent;
 				interrupts = <ICU_GRP_NSR 107 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&cpm_syscon0 1 15>;
+				comreset_u = <0x17>;
+				comwake = <0x6>;
 				status = "disabled";
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				sata-port@0 {
+					reg = <0>;
+					satus = "disabled";
+				};
+				sata-port@1 {
+					reg = <1>;
+					status = "disabled";
+				};
 			};
 
 			cpm_usb3_0: usb3@500000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-slave.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-slave.dtsi
index cdd9bdb..12e3053 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-slave.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-slave.dtsi
@@ -102,12 +102,25 @@
 			};
 
 			cps_sata0: sata@540000 {
-				compatible = "marvell,armada-8k-ahci",
-					     "generic-ahci";
+				compatible = "marvell,armada-cp110-ahci";
 				reg = <0x540000 0x30000>;
+				dma-coherent;
 				interrupts = <ICU_GRP_NSR 107 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&cps_syscon0 1 15>;
 				status = "disabled";
+				comreset_u = <0x17>;
+				comwake = <0x6>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				sata-port@0 {
+					reg = <0>;
+					satus = "disabled";
+				};
+				sata-port@1 {
+					reg = <1>;
+					status = "disabled";
+				};
 			};
 
 			cps_usb3_0: usb3@500000 {
-- 
1.7.9.5

